fix: upgrade google destinations to version 15

shrouti1507 opened the pull request:
## What are the changes introduced in this PR?

Google has got a new API version 15. We are updating API version of 
- Google adwords offline conversion
-  Google adwords enhanced conversion
- Google adwords remarketing lists

Also we are adding support of the new fields added by google for this above destination

Doc references :

- enhanced conversion --> adding job_id https://developers.google.com/google-ads/api/rest/reference/rest/v15/customers/uploadConversionAdjustments#ConversionAdjustment

- google-remarketing-list <— need to add consent field https://developers.google.com/google-ads/api/rest/reference/rest/v15/CustomerMatchUserListMetadata#:~:text=of%20CUSTOMER_MATCH_USER_LIST%20type.-,consent,object%20(Consent),-The%20consent%20setting

- offline conversion 

  - click conversion adding consent support https://developers.google.com/google-ads/api/reference/rpc/v15/ClickConversion#consent
  
  - call conversion https://developers.google.com/google-ads/api/reference/rpc/v15/CallConversion#consent
  
  - store sales consent https://developers.google.com/google-ads/api/reference/rpc/v15/UserData#consent

resolves INT-976

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Type of change

**enhancement**

- Existing capabilities/behavior


- New capabilities/behavior

If the pull request is a **new feature**,

### Any technical or performance related pointers to consider with the change?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.

N/A

### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:

N/A

<hr>

### Developer checklist

- [x] **No breaking changes are being introduced.**

- [x] Are all related docs linked with the PR?

- [x] Are all changes manually tested?

- [x] Does this change require any documentation changes?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

koladilip commented:
Do we need to migrate all at once? is it backward compatible for customers? How did you know about this change?

ItsSudip commented on src/v0/destinations/google_adwords_enhanced_conversions/data/trackConfig.json:
```diff
@@ -129,5 +129,13 @@
+  },
+  {
+    "destKey": "jobId",
+    "sourceKeys": ["properties.jobId"],
```
```suggestion
    "sourceKeys": ["properties.jobId","messageId"],
```
let's add messageId as well as it can be used as a differentiator of events.

shrouti1507 replied:
as we discussed jobId is supposed to be an integer. So we cannot do this . Ref: https://developers.google.com/google-ads/api/rest/reference/rest/v15/customers/uploadConversionAdjustments#ConversionAdjustment:~:text=%3A%20boolean%2C-,%22jobId%22%3A%20integer,-%7D

ItsSudip commented on src/v0/destinations/google_adwords_offline_conversions/transform.js:
```diff
@@ -59,6 +60,10 @@ const getConversions = (message, metadata, { Config }, event, conversionType) =>
 
+  const consentObject = populateConsentForGoogleDestinations(properties);
+  if (Object.keys(consentObject).length > 0) {
+    payload.conversions[0].consent = populateConsentForGoogleDestinations(properties);
```
```suggestion
    payload.conversions[0].consent = consentObject;
```

ItsSudip commented on src/v0/destinations/google_adwords_remarketing_lists/networkHandler.js:
```diff
@@ -91,11 +97,11 @@ const runTheJob = async (endpoint, headers, method, jobId) => {
   // step1: offlineUserDataJobs creation
 
-  const firstResponse = await createJob(endpoint, customerId, listId, headers, method);
+  const firstResponse = await createJob(endpoint, customerId, listId, headers, method, consent);
```
```suggestion
  const firstResponse = await createJob(endpoint, headers, method, params);
```

shrouti1507 replied:
addressed

ItsSudip reviewed (commented):
Why test case files are getting changed so much? Is it due to formatting? Can you add some test cases regarding consent?

shrouti1507 commented:
> Do we need to migrate all at once? is it backward compatible for customers? How did you know about this change?

we are on version 14:

sunset date May 2024

ref: https://developers.google.com/google-ads/api/docs/sunset-dates#timetable

According to their documentation no breaking changes has been introduced

Release notes:

https://developers.google.com/google-ads/api/docs/release-notes#v15

shrouti1507 commented:
> Why test case files are getting changed so much? Is it due to formatting? Can you add some test cases regarding consent?

yes, due to formatting. should I move back to the existing formatting to avoid confusion? This was formatted while performing pre-commit hooks. I have already added that. Added description as well

ItsSudip reviewed (commented):
LGTM

shrouti1507 commented:
we have raised individual PRs for the three google ads destination
